pipeline/frontend/yaml/compiler/params.go

```diff
@@ -66,6 +66,16 @@ func sanitizeParamValue(v interface{}) (string, error) {
 		return fmt.Sprintf("%v", vv.Float()), nil
 
 	case reflect.Map:
+		if fromSecret, ok := v.(map[string]interface{}); ok {
```

Comment:
if secrets != nil

Reply:
Since we are only reading we don't need a nil check. Nil maps behave like empty maps when reading,